File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.js
// Last updated: 4/29/2026, 10:32:54 AM
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let left = 0;
    let right = 0;
    const len = nums.length;
    let sum = nums[0];
    let minLen = Infinity;
    while(right < len) {
        if(sum < target) {
            right++;
            sum += nums[right];
        } else {
            while(sum >= target) {
                const len = right - left + 1;
                if(len < minLen) {
                    minLen = len;
                }
                sum -= nums[left];
                left++;
            }
        }
    }
    if(minLen === Infinity) {
        return 0;
    } else {
        return minLen;
    }
};

File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.py
# Last updated: 4/29/2026, 10:32:57 AM
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        sum = 0
        min_len = None
        for i in range(len(nums)):
            sum += nums[i]
            while sum >= target:
                if min_len == None:
                    min_len = i - left + 1
                else:
                    min_len = min(min_len, i - left + 1)
                sum -= nums[left]
                left += 1
        if min_len == None:
            return 0
        else:
            return min_len